examples/angular2/services/store.ts

```diff
@@ -0,0 +1,81 @@
+/// <reference path="../typings/node-uuid/node-uuid.d.ts" />
+/// <reference path="../typings/storejs/storejs.d.ts" />
+/// <reference path="../typings/angular2/angular2.d.ts" />
+
+import {Injectable} from 'angular2/angular2';
+import * as uuid from 'node-uuid';
+
+export class Todo {
+	completed: Boolean;
+	editing: Boolean;
+	title: String;
+	uid: String;
+	setTitle(title: String) {
+		this.title = title.trim();
+	}
+	constructor(title: String) {
+		this.uid = uuid.v4();
+		this.completed = false;
+		this.editing = false;
+		this.title = title.trim();
+	}
+}
+
+@Injectable()
+export class TodoStore {
+	todos: Array<Todo>;
+	constructor() {
+		let persistedTodos = store.get('angular2-todos') || [];
+		// Normalize back into classes
+		this.todos = persistedTodos.map( (todo: {title: String, completed: Boolean, uid: String}) => {
+			let ret = new Todo(todo.title);
+			ret.completed = todo.completed;
+			ret.uid = todo.uid;
+			return ret;
+		});
+	}
+	_updateStore() {
+		store.set('angular2-todos', this.todos);
+	}
+	get(state: {completed: Boolean}) {
+		return this.todos.filter((todo: Todo) => todo.completed == state.completed);
+	}
+	allCompleted() {
+		return this.todos.length == this.getCompleted().length;
+	}
+	setAllTo(toggler) {
+		this.todos.forEach((t: Todo) => t.completed = toggler.checked);
```

Comment:
as to the use of ( ) here @sindresorhus 

https://github.com/Microsoft/TypeScript/issues/3323#issuecomment-107653842